src/component/term/ParentTermSelector.tsx

```diff
@@ -187,9 +192,8 @@ export class ParentTermSelector extends React.Component<
   };
 
   private onFlatListToggle = () => {
-    this.setState({ flatList: !this.state.flatList }, () =>
-      this.treeComponent.current.resetOptions()
-    );
+    this.props.setTermsFlatList(!this.props.flatList);
+    this.treeComponent.current.resetOptions();
```

Comment:
Won't this cause a double options reset? One here and the other in `componentDidMount` when the `flatList` prop value changes...